### ISSUE TITLE
Backticks are not allowed anymore because of the rewritten `type_name` rule.

### ISSUE THREAD

**Issue by OliverNielsen** [realm/SwiftLint#4571]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

After the newly released 0.50.0 version, the rule `type_name` sees names, which is being escaped using backticks, as errors.
This might be intended behaviour but it was not an issue in (at least) 0.49.1.

##### Complete output when running SwiftLint, including the stack trace and command used

Command used:
```bash
swiftlint
```

I cannot post stack trace because it is a corporate project but this is the error:
```bash
Type Name Violation: Type name should only contain alphanumeric characters: '`Type`' (type_name)
```

### Environment

* SwiftLint version `0.50.0`
* Installed using Homebrew.
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
No.
* Which Xcode version are you using (check `xcodebuild -version`)?
Xcode 14.0
Build version 14A309

Example of rule triggering code:
```swift
// This triggers a violation:
struct SomeStruct {
    enum `Type` {
        case x, y, z
    }
}
```

**Comment by jpsim** [comment]
Fix in #4592
